### DISCUSSION TITLE
Adding response object to a function?

### DISCUSSION THREAD

**Discussion by baradhili** [dedoc/scramble#187] [Q&A]
I have some code like below, I'd like the /docs/api show me the expected response (also below), however I'm not sure how to tell it what the structure should be? I'm trying to follow https://scramble.dedoc.co/usage/response but it doesn't seem to have an effect
```
<?php

namespace App\Http\Controllers;

use App\Models\User;
use Illuminate\Http\Request;
use Illuminate\Validation\Rule;
use Illuminate\Validation\ValidationException;
use Spatie\Permission\Models\Role;
use Spatie\Permission\Models\Permission;
use Illuminate\Support\Facades\Hash;
.
.
.
/**
     * Get the currently authenticated user.
     *
     * @param Request $request
     * @response UserResource
     */
    public function currentUser(Request $request)
    {
        return response()->json(['user' => $request->user()], 200);
    }
```

Actual response
```
{
  "user": {
    "id": 1,
    "name": "Test User",
    "email": "testuser@example.com",
    "email_verified_at": null,
    "ip_address": "192.168.56.1",
    "created_at": "2023-08-19T01:55:20.000000Z",
    "updated_at": "2023-08-19T02:00:13.000000Z"
  }
}
```

What the doc tells me will be the response
```
{
  "user": "string"
}
```

**Comment by romalytvynenko** [comment]
@baradhili I don't see `UserResponse` class used.

Maybe try

```
@response array{user: User}
```

**Reply by baradhili** [reply]
hmm sadly still tells me 
```
{
  "user": "string"
}
```
I will try a more verbose building of the response in the hope that it picks that up

**Comment by baradhili** [comment]
Changing to the below gives me "string" in the doc.. I'm sure I'm missing something small :{

```
/**
     * Get the currently authenticated user.
     *
     * @param Request $request
     * @response array{user: User}
     */
    public function currentUser(Request $request) 
    {
        $user = $request->user();
        $roles = $user->roles;
        $permissions = $user->getAllPermissions();
        $user->roles = $roles;
        $user->permissions = $permissions;
        return response()->json($user, 200);
    }
```

**Reply by romalytvynenko** [reply]
> I'm sure I'm missing something small

This can be a bug as well. Let me check.

Meanwhile, can you try something like this? https://scramble.dedoc.co/usage/response#response-description

```php
<?php
/*...*/
use App\Models\User;
/*...*/

    /**
     * Get the currently authenticated user.
     *
     * @param Request $request
     */
    public function currentUser(Request $request)
    {
        /** 
         * @body array{user: User}
         */
        return response()->json(['user' => $request->user()], 200);
    }
```

**Comment by baradhili** [comment]
I updated the comment as:

    /**
     * Get the currently authenticated user.
     *
     * @param Request $request
     * @return \Illuminate\Http\JsonResponse
     * 
     * @response {
     *   "user": {
     *     "id": "integer",
     *     "name": "string",
     *     "email": "string",
     *     "email_verified_at": "datetime|null",
     *     "ip_address": "string",
     *     "created_at": "datetime",
     *     "updated_at": "datetime",
     *     "roles": "array",
     *     "permissions": "array"
     *   }
     * }
     */
    public function currentUser(Request $request)
    {
        $user = $request->user();
        $user->roles = $user->getRoleNames();
        $user->permissions = $user->getAllPermissions();
        return response()->json(['user' => $user], 200);
    }

It still shows as
```
{
  "user": "string"
}
```

So probably a bug?
